uvicorn/protocols/http/h11_impl.py

```diff
@@ -417,12 +456,16 @@ async def send(self, message):
             if message_type != "http.response.start":
                 msg = "Expected ASGI message 'http.response.start', but got '%s'."
                 raise RuntimeError(msg % message_type)
+            message = cast(HTTPResponseStartEvent, message)
 
             self.response_started = True
             self.waiting_for_100_continue = False
 
             status_code = message["status"]
-            headers = self.default_headers + message.get("headers", [])
+            message_headers = cast(
+                List[Tuple[bytes, bytes]], message.get("headers", [])
```

Comment:
As `headers` field in `message` is an `Iterable`, the operation `+` below is not right.

I'm unsure if we should be doing this or make sure that the operation works with iterables that are not list.

Reply:
I would just change this to `headers: TheHeaderType = [*self.default_headers, *message.get("headers", ())]]  # type: ignore[whatever]` or without the #type: ignore` if the type of message["headers"] is inferred.

Reply:
I like the casting more but either way is fine I think asgiref is too permissive here